examples/3d/hierarchy_fractal.rs

```diff
@@ -0,0 +1,216 @@
+    materials: &mut Assets<StandardMaterial>,
+) -> Option<Entity> {
+    if depth == 0 {
+        return None;
+    }
+
+    let line_length = state.param_b * BASE_LENGTH;
+    let line_width = state.param_b * BASE_WIDTH;
+
+    let mesh = meshes.add(Mesh::from(shape::Box {
+        min_x: 0.0,
+        max_x: line_length,
+        min_y: -line_width,
+        max_y: line_width,
+        min_z: -line_width,
+        max_z: line_width,
+    }));
+
+    let material = materials.add(StandardMaterial {
+        emissive: Color::rgb(
+            palette(
+                state.param_a,
+                PALETTE[0].0,
+                PALETTE[1].0,
+                PALETTE[2].0,
+                PALETTE[3].0,
+            ) * 1.5,
+            palette(
+                state.param_a,
+                PALETTE[0].1,
+                PALETTE[1].1,
+                PALETTE[2].1,
+                PALETTE[3].1,
+            ) * 1.5,
+            palette(
+                state.param_a,
+                PALETTE[0].2,
+                PALETTE[1].2,
+                PALETTE[2].2,
+                PALETTE[3].2,
+            ) * 1.5,
+        ),
+        ..default()
+    });
+
+    let speed = state.param_c;
+    let offset = state.param_a;
+    let axis = state.axis;
+
+    let id = commands
+        .spawn_bundle(PbrBundle {
+            mesh,
+            material,
+            transform: Transform {
+                translation: Vec3::new(parent_offset, 0.0, 0.0),
+                rotation: Quat::from_axis_angle(axis, offset),
+                ..default()
+            },
+            ..default()
+        })
+        .insert(Rotate {
+            speed,
+            offset,
+            axis,
+        })
+        .id();
+
+    let depth = depth - 1;
+    let children: Vec<Entity> = (0..WIDTH)
+        .filter_map(|w| {
+            spawn_recursive(
+                depth,
+                state.mutate(*OFFSETS.get(w).expect("not enough offsets for branch width")),
+                line_length,
+                commands,
+                meshes,
+                materials,
+            )
+        })
+        .collect();
+
+    commands.entity(id).push_children(&children);
+
+    Some(id)
+}
+
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    let mut camera = PerspectiveCameraBundle::new_3d();
+    camera.transform.translation.z = 100.0;
+    commands.spawn_bundle(camera);
+
+    let state = NodeState {
+        axis: Vec3::Z,
+        param_a: 0.0,
+        param_b: 1.0,
+        param_c: 0.04,
+    };
+
+    spawn_recursive(
+        DEPTH,
+        state,
+        0.0,
+        &mut commands,
+        &mut (*meshes),
+        &mut (*materials),
+    );
+}
+
+fn main() {
+    App::new()
+        .insert_resource(ClearColor(Color::BLACK))
+        .insert_resource(WindowDescriptor {
+            width: 1400.0,
+            height: 900.0,
+            ..default()
```

Comment:
I think we should use ..Default::default() here

Reply:
I'm not sure. I think this is fine?

`default()` was recently added and all examples are using it (see #4071)

Reply:
Oh nice, I forgot to pull :)